src/layout/Border.js

```diff
@@ -218,115 +199,164 @@
         return null;
     };
 
-    Border.prototype.getSize = function(i){
-        switch(this.sectionTypes()[i]){
-            case "top":return this.topSize();
-            case "right":return this.rightSize();
-            case "bottom":return this.bottomSize();
-            case "left":return this.leftSize();
+    Border.prototype.setLayoutOffsets = function () {
+        this._offsetX = this._element.node().getBoundingClientRect().left + (this.gutter()/2);
+        this._offsetY = this._element.node().getBoundingClientRect().top + (this.gutter()/2);
+    };
+
+    Border.prototype.dragStart = function(handle){
+        d3.event.sourceEvent.stopPropagation();
+        var context = this;
+
+        this._dragCell = handle;
+        this._dragCellStartSize = this[handle+"Size"]();
+        
+        if(this[handle+"ShrinkWrap"]()){
+            this[handle+"Percentage"](0);
+            this[handle+"ShrinkWrap"](false);
         }
+        
+        var handleElm = d3.select("#" + context.id() + " > div.borderHandle_"+handle);
+        context._handleTop = parseFloat(handleElm.style("top").split("px")[0]);
+        context._handleLeft = parseFloat(handleElm.style("left").split("px")[0]);
+        
+        this._dragPrevX = d3.event.sourceEvent.clientX;
+        this._dragPrevY = d3.event.sourceEvent.clientY;
     };
-    Border.prototype.changeSize = function(i,delta){
-        switch(this.sectionTypes()[i]){
+    Border.prototype.dragTick = function(handle){
+        var context = this;
+        
+        var xDelta = this._dragPrevX - d3.event.sourceEvent.clientX;
+        var yDelta = this._dragPrevY - d3.event.sourceEvent.clientY;
+
+        switch(handle){
             case "top":
-                this.topSize(this.topSize() + delta);
+                if(yDelta !== 0){
+                    _moveHandles(handle,yDelta);
+                }
                 break;
             case "right":
-                this.rightSize(this.rightSize() + delta);
+                if(xDelta !== 0){
+                    _moveHandles(handle,xDelta);
+                }
                 break;
             case "bottom":
-                this.bottomSize(this.bottomSize() + delta);
+                if(yDelta !== 0){
+                    _moveHandles(handle,yDelta);
+                }
                 break;
             case "left":
-                this.leftSize(this.leftSize() + delta);
-                break;
-            case "center":
-                this.centerSize(this.centerSize() + delta);
+                if(xDelta !== 0){
```

Comment:
You might want to move this ```!== 0``` test into _moveHandles?  Would clean up this code a little bit?